indexing-service/src/main/java/io/druid/indexing/common/task/AbstractTask.java

```diff
@@ -28,15 +28,19 @@
 import io.druid.indexing.common.actions.LockListAction;
 import io.druid.query.Query;
 import io.druid.query.QueryRunner;
-import java.util.Map;
 import org.joda.time.DateTime;
 import org.joda.time.Interval;
 
 import java.io.IOException;
+import java.util.Map;
 
 public abstract class AbstractTask implements Task
 {
   private static final Joiner ID_JOINER = Joiner.on("_");
+  protected static final int REALTIME_TASK_PRIORITY = 100;
+  protected static final int INDEX_TASK_PRIORITY = 75;
+  protected static final int MERGE_TASK_PRIORITY = 50;
+  protected static final int DEFAULT_TASK_PRIORITY = 25;
```

Comment:
Yes I remember that however I want some clarification on two points -
1. For all tasks than Realtime, Index and Merge we want the priority to be 0. Thus, realtime, index and merge tasks will have some priority > 0. Right ?
2. As of now if the priority is set to 0 in task json then the task takes the default priority as per the task type not 0. So, if one submits a realtime task with priority set to 0, the actual priority will be 100. Do we want to change this behavior as it may not go well with first approach ?
So, the confusion is what semantics we want here 0 to be taken as a literal number or to say take the default value for the task type ?
Anyways, any valid integer value will work for priority, so negative values will work fine.